x-pack/plugins/task_manager/server/task_pool.ts

```diff
@@ -44,8 +45,10 @@ export class TaskPool {
    * @prop {Logger} logger - The task manager logger.
    */
   constructor(opts: Opts) {
-    this.maxWorkers = opts.maxWorkers;
     this.logger = opts.logger;
+    opts.maxWorkers$.subscribe((maxWorkers) => {
+      this.maxWorkers = maxWorkers;
+    });
```

Comment:
This makes me wonder if we should extract the worker "tracking" from the pool as the whole reactive paradigm doesn't sit well with classes and state (subscribe to update an internal field kind of breaks the point of the reactiveness, but that would be a wider refactor... it doesn't need to be done in this PR 🤷 )